examples/http-client/src/main.rs

```diff
@@ -0,0 +1,109 @@
+#![no_main]
+#![no_std]
+#![feature(impl_trait_in_assoc_type)]
+#![feature(type_alias_impl_trait)]
+#![feature(used_with_arg)]
+
+use ariel_os::{
+    debug::log::*,
+    network,
+    reexports::{embassy_net, embassy_time},
+};
+use embassy_net::{
+    dns::DnsSocket,
+    tcp::client::{TcpClient, TcpClientState},
+};
+use embassy_time::{Duration, Timer};
+use reqwless::{
+    client::{HttpClient, TlsConfig, TlsVerify},
+    request::Method,
+};
+
+// RFC8449: TLS 1.3 encrypted records are limited to 16 KiB + 256 bytes.
+const MAX_ENCRYPTED_TLS_13_RECORD_SIZE: usize = 16640;
+// Required by `embedded_tls::TlsConnection::new()`.
+const TLS_READ_BUFFER_SIZE: usize = MAX_ENCRYPTED_TLS_13_RECORD_SIZE;
+// Can be smaller than the read buffer (could be adjusted: trade-off between memory usage and not
+// splitting large writes into multiple records).
+const TLS_WRITE_BUFFER_SIZE: usize = 4096;
+
+const TCP_BUFFER_SIZE: usize = 1024;
+const HTTP_BUFFER_SIZE: usize = 1024;
+
+const MAX_CONCURRENT_CONNECTIONS: usize = 2;
+
+// Endpoint to send the GET request to.
+const ENDPOINT_URL: &str = env!("ENDPOINT_URL");
+
+#[ariel_os::config(network)]
+const NETWORK_CONFIG: embassy_net::Config = {
+    use embassy_net::Ipv4Address;
+
+    embassy_net::Config::ipv4_static(embassy_net::StaticConfigV4 {
+        address: embassy_net::Ipv4Cidr::new(Ipv4Address::new(10, 42, 0, 61), 24),
+        dns_servers: heapless::Vec::new(),
+        gateway: Some(Ipv4Address::new(10, 42, 0, 1)),
+    })
+};
+
+#[ariel_os::task(autostart)]
+async fn main() {
+    let stack = network::network_stack().await.unwrap();
+
+    let tcp_client_state =
```

Comment:
... and these lines seem to be identical for a tls http client, with *maybe* configurable buffer sizes. I guess for now we're OK but we'll need to abstract this.

Reply:
would you move it into a function already? and moce that to the bottom of the file?